### PR TITLE
gitfox: update livecheck

### PR DESCRIPTION
**Pull request by khipp**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

Related to https://github.com/Homebrew/homebrew-cask/issues/171006.

### CODE REVIEW

File: Casks/g/gitfox.rb [hunk 1]
```diff
@@ -1,16 +1,18 @@
 cask "gitfox" do
-  version "3.9.3,9610"
+  version "9610"
```
Comment (chenrui333) [inline Casks/g/gitfox.rb:2]: but 9610 is build number? do we really want to drop 3.9.3?

Comment (khipp) [reply Casks/g/gitfox.rb]: I added the version information back in for clarity.

File: Casks/g/gitfox.rb [hunk 2]
```diff
@@ -9,12 +9,14 @@
   homepage "https://www.gitfox.app/"
 
   livecheck do
-    url "https://storage.googleapis.com/gitfox/Gitfox.latest.stable.zip"
-    strategy :extract_plist
+    url "https://api.gitfox.app/v1/versions"
+    strategy :json do |json|
+      json["response"]["builds"].map { |item| "#{item["short_version_string"]},#{item["build"]}" }
```
Comment (samford) [inline Casks/g/gitfox.rb:14]:
```suggestion
      json.dig("response", "builds")&.map do |item|
        next unless item["short_version_string"] && item["build"]

        "#{item["short_version_string"]},#{item["build"]}"
      end
```

Using `#dig` for multi-level values allows us to use the safe navigation operator after it. This suggestion also accounts for potential `nil` values in the `#map` block, so we don't end up with partial versions (e.g., `,1234`, `1.2.3,`, `,`) if expected values are missing.

Comment (khipp) [reply Casks/g/gitfox.rb]: Thanks!

**Review by krehel — APPROVED** [review]

**Pull request merged**